File: nextcart_frontend/src/services/authService.js
import apiClient from './apiClient'
import { ENDPOINTS } from '@/constants/api'

const authService = {
  register: async (firstName, lastName, email, password) => {
    const res = await apiClient.post(ENDPOINTS.AUTH.REGISTER, { firstName, lastName, email, password })
    return res.data
  },

  login: async (email, password) => {
    const res = await apiClient.post(ENDPOINTS.AUTH.LOGIN, { email, password })
    return res.data
  },

  logout: (refreshToken) =>
    apiClient.post(ENDPOINTS.AUTH.LOGOUT, { refreshToken }),

  refreshTokens: async (refreshToken) => {
    const res = await apiClient.post(ENDPOINTS.AUTH.REFRESH, { refreshToken })
    return res.data
  },

  getMe: async () => {
    const res = await apiClient.get(ENDPOINTS.AUTH.ME)
    return res.data
  },

  updateProfile: async ({ firstName, lastName, phone }) => {
    const res = await apiClient.patch(ENDPOINTS.AUTH.UPDATE_PROFILE, { firstName, lastName, phone })
    return res.data
  },

  changePassword: (currentPassword, newPassword) =>
    apiClient.patch(ENDPOINTS.AUTH.CHANGE_PASSWORD, { currentPassword, newPassword }),

  /** @returns {Promise<{type: string, message: string}>} type is "EMAIL_SENT" or "SOCIAL_ACCOUNT" */
  forgotPassword: async (email) => {
    const res = await apiClient.post(ENDPOINTS.AUTH.FORGOT_PASSWORD, { email })
    return res.data
  },

  resetPassword: async (token, newPassword) => {
    const res = await apiClient.post(ENDPOINTS.AUTH.RESET_PASSWORD, { token, newPassword })
    return res.data
  },
}

export default authService

File: nextcart_backend/src/main/java/com/mhjoy/nextcart/auth/service/AuthService.java
package com.mhjoy.nextcart.auth.service;

import com.mhjoy.nextcart.auth.dto.request.ChangePasswordRequestDto;
import com.mhjoy.nextcart.auth.dto.request.CustomerRegisterRequestDto;
import com.mhjoy.nextcart.auth.dto.request.ForgotPasswordRequestDto;
import com.mhjoy.nextcart.auth.dto.request.LoginRequestDto;
import com.mhjoy.nextcart.auth.dto.request.RefreshTokenRequestDto;
import com.mhjoy.nextcart.auth.dto.request.ResetPasswordRequestDto;
import com.mhjoy.nextcart.auth.dto.request.UpdateProfileRequestDto;
import com.mhjoy.nextcart.auth.dto.response.ForgotPasswordResponseDto;
import com.mhjoy.nextcart.auth.dto.response.TokenResponseDto;
import com.mhjoy.nextcart.auth.dto.response.UserResponseDto;

public interface AuthService {

    UserResponseDto register(CustomerRegisterRequestDto requestDto);

    TokenResponseDto login(LoginRequestDto requestDto);

    TokenResponseDto refresh(RefreshTokenRequestDto requestDto);

    void logout(String refreshToken);

    UserResponseDto me(Long userId);

    UserResponseDto updateMe(Long userId, UpdateProfileRequestDto requestDto);

    void changePassword(Long userId, ChangePasswordRequestDto requestDto);

    /**
     * Handles forgot-password requests.
     *
     * <ul>
     *   <li>Normal user → sends reset-link email, returns {@code EMAIL_SENT}</li>
     *   <li>Google-only user → sends social-login-reminder email, returns {@code SOCIAL_ACCOUNT}</li>
     *   <li>Unknown email → returns {@code EMAIL_SENT} (no email sent; prevents enumeration)</li>
     * </ul>
     */
    ForgotPasswordResponseDto forgotPassword(ForgotPasswordRequestDto requestDto);

    /**
     * Validates the reset token, updates the password, and invalidates the token.
     */
    void resetPassword(ResetPasswordRequestDto requestDto);

    /**
     * Saves a new S3 profile picture key for the user, returning the old key so the
     * caller can delete the previous file from S3 if needed.
     *
     * @return the old profilePictureKey (may be null if none existed)
     */
    String updateProfilePictureKey(Long userId, String newKey);
}
